File: cs50/pset2/caesar.c
#include <stdio.h>
#include <cs50.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        return 1;
    }

    int key = atoi(argv[1]) % 26;
    string text = get_string("Enter text: ");

    for (int i = 0; text[i] != '\0'; i++)
    {
        char target = text[i];

        if (target >= 'a' && target <= 'z')
        {
            target = (target + key <= 'z') ? target + key : 'a' + (target + key - 'z' - 1);
        }
        else if (target >= 'A' && target <= 'Z')
        {
            target = (target + key <= 'Z') ? target + key : 'A' + (target + key - 'Z' - 1);
        }
        text[i] = target;
    }

    printf("ciphertext: %s\n", text);
}

File: cs50/pset6/caesar/caesar.py
from cs50 import get_string
import sys

# make sure the cipher has been given on command line
if len(sys.argv) != 2:
    sys.exit(1)

# covert the passed argument to integer
key = int(sys.argv[1]) % 26
# prompt user for text
text = get_string("Enter text: ")

# some useful constants
a = ord('a')
z = ord('z')
A = ord('A')
Z = ord('Z')

ciphertext = ""

# parse through text characters and encode them if alphabetic
for c in text:
    # convert c to ASCII value for
    # easy comparison and calculation
    target = ord(c)

    if (target >= a and target <= z):
        target -= a
        target += key
        target %= 26
        target += a
    elif (target >= A and target <= Z):
        target -= A
        target += key
        target %= 26
        target += A

    # convert the ASCII code to char
    ciphertext += chr(target)

print(f"ciphertext: {ciphertext}")
